x-pack/metricbeat/module/aws/ec2/ec2.go

```diff
@@ -104,17 +104,18 @@ func (m *MetricSet) Fetch(report mb.ReporterV2) {
 
 		for _, instanceID := range instanceIDs {
 			metricDataQueries := constructMetricQueries(listMetricsOutput, instanceID, m.PeriodInSec)
-			if len(metricDataQueries) == 0 {
-				continue
-			}
 
-			// Use metricDataQueries to make GetMetricData API calls
-			metricDataOutput, err := aws.GetMetricDataResults(metricDataQueries, svcCloudwatch, startTime, endTime)
-			if err != nil {
-				err = errors.Wrap(err, "GetMetricDataResults failed, skipping region "+regionName+" for instance "+instanceID)
-				m.logger.Error(err.Error())
-				report.Error(err)
-				continue
+			// If metricDataQueries, still needs to createCloudWatchEvents.
+			metricDataOutput := []cloudwatch.MetricDataResult{}
+			if len(metricDataQueries) != 0 {
```

Comment:
Can it be negative?

Reply:
I don't think so 🤔 it's looking at the length of an array so 0 or larger.